### ISSUE TITLE
some solid documentation and examples

### ISSUE THREAD

**Issue by verybigelephants** [addyosmani/getUserMedia.js#49]
hello, i don't mean to hate but there is no documentation about the options and the example file demo.js is a complete mess. i think this is what this project really needs

**Comment by addyosmani** [comment]
Thanks for the feedback. This project was started ~4 or so years ago and I currently only have time to review maintenance PRs. If anyone using gUM.js is interested in submitting PRs for better docs or examples, I'm more than happy to review those changes.

**Comment by verybigelephants** [comment]
i am currently using this great library for a work project, once i finish and fully understand it, i will be more than happy to write some helping docs!

edit: i ended up not using it altogether because the whole plugin was too much of a mess and i just gave up and thrown the "update your browser message". as an example, one of the most serious issues i've had is that in case of flash fallback success and failure the callbacks never fired

**Comment by osdiab** [comment]
agreed, I have no idea what some of the flags do (`extern`?), nor when events like `onCapture` occur... etc. I can help you out with that if you're still interested in filling in the gaps @verybigelephants

**Comment by verybigelephants** [comment]
i have decided not to use it. the options didn't really seem to do anything and the events never triggered; i found myself hacking the library and figuring out how to actually use it more than spending time on the actual project :(

so i just convinced our client to tell the users not to use internet explorer or safari. those browsers should just die out anyways
